crates/lint/src/sol/macros.rs

```diff
@@ -1,31 +1,99 @@
+#[macro_export]
+macro_rules! link {
```

Comment:
can you extract this and the other changes not related to the example feature in another PR? so we can merge that right away